### PR TITLE
Pub 1095 post application to backend

### PR DESCRIPTION
**Pull request by joshblackmoor**
### JIRA link ###

https://tools.hmcts.net/jira/browse/PUB-1095

### Change description ###

Added the ability for details from the create media account page to be posted into the API

Superagent has been used over Axios as was having issues posting multipart/form-data to the API, there are examples of where this has been done previously for uploading publications.

For uploading the image I have used Multer and have used the same directory/config that the manual upload uses but this could be changed if agreed we want it to be separate. 

The UI screen hasn't really changed but some screenshots: 

Screenshot 2022-05-11 at 14 13 46

Screenshot 2022-05-11 at 14 14 01

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x ] No
```

### CODE REVIEW

**Review by furnivall — DISMISSED** [review]
Looking good !

File: src/main/resources/requests/accountManagementRequests.ts [hunk 1]
```diff
@@ -1,8 +1,12 @@
-import { accountManagementApi } from './utils/axiosConfig';
+import {accountManagementApi, getAccountManagementCredentials} from './utils/axiosConfig';
 import { Logger } from '@hmcts/nodejs-logging';
 
+const superagent = require('superagent');
 const logger = Logger.getLogger('requests');
+
 export class AccountManagementRequests {
+  public accountManagementUrl = process.env.ACCOUNT_MANAGEMENT_URL || 'https://pip-account-management.staging.platform.hmcts.net/';
+
```
Comment (junaidiqbalmoj) [inline src/main/resources/requests/accountManagementRequests.ts:9]: Why can't we use accountManagementApi from axiosConfig? Is there a special reason of creating accountManagementUrl?

Comment (joshblackmoor) [reply src/main/resources/requests/accountManagementRequests.ts:9]: The accountManagementApi in axiosConfig is creating a new instance of axios with the config we have set I believe and the accountManagementUrl I have added is for use with Superagent, there is another example of where we've done this in dataManagementRequests.ts. If you would prefer I refactor this and grab the consts from axiosConfig instead  then can do :)

Comment (junaidiqbalmoj) [reply src/main/resources/requests/accountManagementRequests.ts:9]: No, that's fine. You can user Superagent. Api staging url is hardcoded in both accoutManagementRequest.ts and axiosConfig file as well. We should keep this hardcoded url in one place only. Can you get this URL from axiosConfig file? Something like:

In axiosConfig file:
export const accountManagementApiUrl = "https://pip-account-management.staging.platform.hmcts.net/";

and use this const in accountManagementRequest file.

Comment (joshblackmoor) [reply src/main/resources/requests/accountManagementRequests.ts:8]: Done 👍

File: src/main/service/createAccountService.ts [hunk 2]
```diff
@@ -117,18 +113,39 @@ export class CreateAccountService {
...
+
+  public removeFile(file: File): void {
+    const filePath = './manualUpload/tmp/' + file['originalname'];
+    try {
+      fs.unlinkSync(filePath);
+    } catch (err) {
+      console.error('Error while deleting ' + file['originalname']);
+    }
+  }
+
```
[the remark this answers is not in the document]
Comment (joshblackmoor) [reply src/main/service/createAccountService.ts]: Done :)

**Review by kurt-uk — APPROVED** [review]
nice work

**Review by ChrisS1512 — DISMISSED** [review]

**Review by furnivall — DISMISSED** [review]

**Review by junaidiqbalmoj — DISMISSED** [review]
LGTM. Nice work :)

**Review by furnivall — APPROVED** [review]
looks good to me!

**Review by ChrisS1512 — APPROVED** [review]

**Pull request merged**